specs/ClientCertificate.md

```diff
@@ -0,0 +1,500 @@
+  /// Display the default client certificate selection dialog prompt to allow the user
+  /// to choose a certificate. The SelectedCertificate property is ignored in this case.
+  COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE_CONTINUE_WITH_DEFAULT_DIALOG,
+} COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE;
+
+[v1_enum] typedef enum COREWEBVIEW2_CLIENT_CERTIFICATE_KIND {
+  /// Specifies smart card certificate.
+  COREWEBVIEW2_CLIENT_CERTIFICATE_KIND_SMART_CARD,
+  /// Specifies PIN certificate.
+  COREWEBVIEW2_CLIENT_CERTIFICATE_KIND_PIN,
+  /// Specifies other certificate.
+  COREWEBVIEW2_CLIENT_CERTIFICATE_KIND_OTHER,
+} COREWEBVIEW2_CLIENT_CERTIFICATE_KIND;
+
+[uuid(92d41f20-8e92-11eb-8dcd-0242ac130003), object, pointer_default(unique)]
+interface ICoreWebView2_3 : ICoreWebView2_2 {
+  /// Add an event handler for the ClientCertificateRequested event.
+  /// The ClientCertificateRequested event is raised when the WebView2
+  /// is making a request to an HTTP server that needs a client certificate
+  /// for HTTP authentication and when there is atleast one mutually trusted certitifcate.
+  /// Read more about HTTP client certificates at
+  /// [RFC 8446 The Transport Layer Security (TLS) Protocol Version 1.3](https://tools.ietf.org/html/rfc8446).
+  ///
+  /// With this event you have several options for responding to client certificate requests:
+  ///
+  /// *  You can query the mutually trusted CA certificates, and select a certificate
+  /// from the list to respond to the server.
+  /// *  You can choose to respond to the server without a certificate.
+  /// *  You can choose to display default client certificate selection dialog prompt
+  /// to let user to respond to the server.
+  /// *  You can cancel the request.
+  ///
+  /// If you doesn't handle the event, WebView2 will
+  /// show the default client certificate selection dialog prompt to user.
+  ///
+  /// \snippet SettingsComponent.cpp ClientCertificateRequested
+  HRESULT add_ClientCertificateRequested(
+      [in] ICoreWebView2ClientCertificateRequestedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove an event handler previously added with add_ClientCertificateRequested.
+  HRESULT remove_ClientCertificateRequested([in] EventRegistrationToken token);
+}
+
+/// Provides access to the certificate metadata
+[uuid(9ca02116-8e92-11eb-8dcd-0242ac130003), object, pointer_default(unique)]
+interface ICoreWebView2ClientCertificate : IUnknown {
+  /// Subject of the certificate.
+  [propget] HRESULT Subject([out, retval] LPWSTR* subject);
+  /// Name of the certificate authority that issued the certificate.
+  [propget] HRESULT Issuer([ out, retval ] LPWSTR* issuer);
+  /// The valid start date and time for the certificate as the number of seconds since
+  /// the UNIX epoch.
+  [propget] HRESULT ValidStartDate([out, retval] double* validStartDate);
```

Comment:
Is this the first time we've needed to express a date/time in WebView2? If not, do the other WebView2 date/time properties/parameters also use "seconds in unix epoch as a double"?

Windows Runtime calls these properties ValidFrom and ValidTo.